R/ode.R

```diff
@@ -82,7 +82,15 @@ daedalus_rhs <- function(t, state, parameters) {
 
   scaling <- if (switch) openness else 1.0
   r0_econ <- r0 * scaling
-  r0 <- r0 * mean(scaling) # as otherwise no scaling on r0
+
+  # create empty array of the dimensions of state
+  d_state <- array(0.0, dim(state_))
+
+  # get new deaths and implement social distancing only when closures are active
+  # as described in https://github.com/robj411/p2_drivers
+  d_state[, i_D, ] <- omega * state_[, i_H, ]
+  new_deaths <- sum(d_state[, i_D, ])
+  r0 <- r0 * ifelse(switch, get_distancing_coefficient(new_deaths), 1.0)
```

Comment:
```suggestion
  r0 <- r0 * if (switch) get_distancing_coefficient(new_deaths) else 1.0
```

`ifelse` is vectorised, and also full of footguns